Automattic/newspack-plugin
# Add/proxy improvements

### All Submissions:






### Changes proposed in this Pull Request:



1. Enables override of a proxy URL. This is needed in environments where `NEWSPACK_*_PROXY` values are set globally and can't be overridden by setting a value in `wp-config.php`.
2. Adds a `\Newspack\Logger` class for logging stuff. This will be useful for debugging live sites.

### How to test the changes in this Pull Request:

1. Set the `NEWSPACK_GOOGLE_OAUTH_PROXY_OVERRIDE` environment and observe it does override the `NEWSPACK_GOOGLE_OAUTH_PROXY` value
2. Observe site logs (by `tail`ing the `wp-content/debug.log` or the PHP error log, depending on configuration)
3. Observe nothing interesting is logged by default when going through the Google OAuth flow
4. Set the `NEWSPACK_LOG_LEVEL` environment variable to `1`
5. Observe lines logged with `[NEWSPACK]` prefix when refreshing the Google OAuth token*
6. Set the `NEWSPACK_LOG_LEVEL` variable to `2`
7. Observe the logged lines contain the name of the class and method which called the logger

\* refresh can be triggered by changing the `expires_at` value in the value of `_newspack_google_oauth` option after going through the Google OAuth flow

### Other information:
